plc4j/spi/src/main/java/org/apache/plc4x/java/spi/configuration/ConfigurationFactory.java

```diff
@@ -178,63 +171,78 @@ private static Object toFieldValue(Field field, String valueString) {
             try {
                 ConfigurationParameterConverter<?> converter = converterClass.getDeclaredConstructor().newInstance();
                 if (converter.getType().isAssignableFrom(field.getType())) {
-                    return converter.convert(valueString);
+                    valueString = converter.convert(valueString).toString();
```

Comment:
IntelliJ is reporting that this value is never used. This is due to the fact that in line 179 you are always throwing an exception.